### PR TITLE
Sharding s3 key names and removing some trash

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Virtual filesystem over s3 (experimental feature) will use random object names split into multiple path prefixes for better performance on AWS.

### CODE REVIEW

File: src/Common/getRandomASCIIString.h [hunk 1]
```diff
@@ -1,10 +1,12 @@
 #pragma once
+
 #include <Core/Types.h>
 
+
 namespace DB
 {
-/// Slow random string. Useful for random names and things like this. Not for
-/// generating data.
-String getRandomASCIIString(size_t len = 32, char first = 'a', char last = 'z');
```
Comment (alexey-milovidov) [inline src/Common/getRandomASCIIString.h:8]: Trash.

File: src/Common/getRandomASCIIString.cpp [hunk 2]
```diff
@@ -5,10 +5,11 @@
 namespace DB
 {
 
-String getRandomASCIIString(size_t len, char first, char last)
+String getRandomASCIIString(size_t length)
 {
-    std::uniform_int_distribution<int> distribution(first, last);
-    String res(len, ' ');
```
Comment (alexey-milovidov) [inline src/Common/getRandomASCIIString.cpp:11]: Trash.

**Review by kitaisreal — APPROVED** [review]

File: src/Disks/S3/DiskS3.cpp [hunk 3]
```diff
@@ -247,8 +242,17 @@ std::unique_ptr<WriteBufferFromFileBase> DiskS3::writeFile(const String & path,
...
+    /// Path to store the new S3 object.
+
+    /// Total length is 32 a-z characters for enough randomness.
+    /// First 3 characters are used as a prefix for
+    /// https://aws.amazon.com/premiumsupport/knowledge-center/s3-object-key-naming-pattern/
+
+    constexpr size_t key_name_total_size = 32;
+    constexpr size_t key_name_prefix_size = 3;
+
+    auto s3_path = fmt::format("{}/{}",
```
Comment (alesapin) [inline src/Disks/S3/DiskS3.cpp:254]: So we will have "random" prefixes for all objects?

Comment (alexey-milovidov) [reply src/Disks/S3/DiskS3.cpp]: Yes, we simply put `/` after first three random letters.

**Comment by alexey-milovidov** [comment]
#40968

**Comment by alexey-milovidov** [comment]
This PR was split to two parts and merged.

**Pull request closed without merging**